internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -676,6 +677,34 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				},
 			},
 
+			"bootstrap_profile": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Computed: true,
```

Comment:
Could you add a note here with a brief explanation of why this is Optional + Computed? ([ref](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/best-practices.md#setting-properties-to-optional--computed))

While not all O+C properties currently have this in the provider, we've started to add the context to new O+C properties in case we need to revisit it in the future.

```suggestion
				Optional: true,
				// Note: O+C because <reason>
				Computed: true,
```